app/models/qernel/merit_facade/hybrid_offshore_adapter.rb

```diff
@@ -0,0 +1,182 @@
+        @converter_input_capacity = (
+          converter_node.node_api.input_capacity *
+          converter_node.node_api.number_of_units *
+          converter_node.node_api.availability
+        )
+
+        @clearing = Array.new(8760, 0.0)
+        @curtailed = Array.new(8760, 0.0)
+
+        @converter_curve = Qernel::Causality::ActiveLazyCurve.new do |frame|
+          @clearing[frame] * -1
+        end
+      end
+
+      def inject!
+        super
+
+        total_demand = participant.production
+        curtailment_demand = @curtailed.sum * 3600
+
+        calculate_curves!
+
+        # The curtailment egde is a share edge, so we calculate the share
+        find_edge(curtailment_node).dataset_set(:share, safe_div(curtailment_demand, total_demand))
+
+        # The converter node is a constant edge, so we set the demand directly
+        find_edge(converter_node).dataset_set(:share, @converter_curve.sum * 3600)
+
+        output_node.dataset_lazy_set(:electricity_input_curve) do
+          @context.curves.derotate(@output_curve)
+        end
+
+        curtailment_node.dataset_lazy_set(:electricity_input_curve) do
+          @context.curves.derotate(@curtailed)
+        end
+
+        input_node.dataset_lazy_set(:electricity_output_curve) do
+          @context.curves.derotate(@input_curve)
+        end
+      end
+
+      private
+
+      def find_edge(consumer, carrier = @context.carrier)
+        edge = target_api.output_edges.find { |e| e.lft_node == consumer }
+
+        unless edge
+          raise "Couldn't find a #{carrier.inspect} edge between #{target_api.key} " \
+                "and #{consumer.key}"
+        end
+
+        edge
+      end
+
+      def safe_div(num, denom)
+        num.zero? || denom.zero? ? 0.0 : num / denom
+      end
+
+      def producer_class
+        Merit::ConstrainedVolatileProducer
+      end
+
+      def producer_attributes
+        attrs = super
+
+        attrs[:constraint] = ->(point, amount) { constrain(point, amount) }
+
+        attrs
+      end
+
+      # Calculates the energy that is constrained from going on the market. The constrained
+      # energy is located to other components.
+      #
+      # Returns the energy that is allowed to go on the market
+      def constrain(point, amount)
+        converted = to_converter(amount)
+        @clearing[point] = converted
+
+        to_market = [amount - converted, @output_input_capacity].min
+        @curtailed[point] = amount - converted - to_market
+
+        to_market
+      end
+
+      # The amount that flows to the converter in that hour
+      def to_converter(amount)
+        (amount - @output_input_capacity).clamp(0.0, @converter_input_capacity)
+      end
+
+      # Caluclate the curves after Merit calculations have finished
+      def calculate_curves!
+        @output_curve = Array.new(8760, 0.0)
+        @input_curve = Array.new(8760, 0.0)
+        @converter_curve = Array.new(8760, 0.0)
+
+        8760.times do |frame|
```

Comment:
A similar question could be asked for the global constant HOURS_PER_YEAR. We recently got the question whether it would be possible to simulate leap years. I know this is very unlikely to happen, but wouldn't it be easier if this is a global variable throughout the model? What are reasons to use 8760 hardcoded?